src/main/java/org/spongepowered/api/service/context/ContextCalculator.java

```diff
@@ -22,36 +22,37 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.service.permission.context;
-
-import org.spongepowered.api.service.permission.Subject;
+package org.spongepowered.api.service.context;
 
 import java.util.Set;
 
 /**
- * Calculate the availability of contexts. These methods may be invoked
- * frequently, and therefore should be fast.
+ * Calculate the availability of {@link Context}s for a {@link Contextual}. 
+ * These methods may be invoked frequently, and therefore should be fast.
+ * 
  */
 public interface ContextCalculator {
```

Comment:
Because a ContextCalculator inspects the Contextual object.

Let's say a ``WorldContextCalculator`` for a ``Subject``:

Get the type of the subject (instanceof Player, User, Console, etc.).
If Subject is located in a world, get the world (getLocation().getExtent()...)

A ``WorldContextCalculator`` for an ``Account``:
Check if the Account belongs to a player (instanceof Wallet). Get the player. Get the world of the player.